File: reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        
        int reverse = 0;
        
        while (x != 0) {
            
            int newDigit = x % 10;
            
            int temp = (reverse * 10) + newDigit;
            
            if ((temp - newDigit) / 10 != reverse) { return 0; }
            
            reverse = temp;
                
            x /= 10;
        }
        
        return reverse;
    }
}

File: reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        
        if (x < INT_MIN || x > INT_MAX) { return 0; }
        
        long int reverseInt = 0;
        
        int newDigit, temp;
        
        while (x != 0) {
            
            newDigit = x % 10;
            
            temp = (reverseInt * 10) + newDigit;
            
            if ((temp - newDigit) / 10 != reverseInt) { return 0; }
            
            reverseInt = temp;
            
            x /= 10;
        }
        
        return (int)reverseInt;
    }
};